assets/src/pages/room/components/StreamPlayer/PinnedTilesSection.tsx

```diff
@@ -11,48 +11,58 @@ import {
   showSimulcast?: boolean;
 };
 
-const PinnedTilesSection: FC<Props> = ({ pinnedTile, unpin, webrtc, showSimulcast }: Props) => {
-  const tileType = pinnedTile.typeName;
-  const hasInitials = tileType === "local" || tileType === "remote";
-
+const PinnedTilesSection: FC<Props> = ({ pinnedTiles, unpin, webrtc, showSimulcast }: Props) => {
+  const gridConfig = getGridConfig(pinnedTiles.length);
+  const style = clsx(gridConfig.columns, gridConfig.grid, gridConfig.gap, gridConfig.padding, gridConfig.rows);
   return (
-    <div className="active-screensharing-grid h-full grid-cols-1">
-      <MediaPlayerTile
-        key={pinnedTile.mediaPlayerId}
-        peerId={pinnedTile.peerId}
-        video={pinnedTile.video}
-        audio={pinnedTile.typeName === "remote" ? pinnedTile.audio : null}
-        streamSource={pinnedTile.streamSource}
-        blockFillContent={pinnedTile.typeName === "screenShare"}
-        flipHorizontally={pinnedTile.typeName === "local"}
-        layers={
-          <>
-            {hasInitials && showDisabledIcon(pinnedTile.video) ? (
-              <InitialsImage initials={pinnedTile.initials} />
-            ) : null}
-            <PeerInfoLayer
-              topRight={<PinIndicator />}
-              topLeft={
-                hasInitials && showDisabledIcon(pinnedTile.audio) ? (
-                  <DisabledMicIcon isLoading={isLoading(pinnedTile.audio)} />
-                ) : null
+    <div className={"grid h-full w-full auto-rows-fr gap-3"}>
+      <div className={clsx("h-full w-full", style)}>
+        {pinnedTiles.map((pinnedTile: MediaPlayerTileConfig) => {
+          const tileType = pinnedTile.typeName;
+          const hasInitials = tileType === "local" || tileType === "remote";
+          return (
+            <MediaPlayerTile
+              key={pinnedTile.mediaPlayerId}
+              className={clsx(gridConfig.span, gridConfig.tileClass)}
+              peerId={pinnedTile.peerId}
+              video={pinnedTile.video}
+              audio={tileType === "remote" ? pinnedTile.audio : null}
+              streamSource={pinnedTile.streamSource}
+              blockFillContent={tileType === "screenShare"}
+              flipHorizontally={tileType === "local"}
+              layers={
+                <>
+                  {hasInitials && showDisabledIcon(pinnedTile.video) ? (
+                    <InitialsImage initials={pinnedTile.initials} />
+                  ) : null}
+                  <PeerInfoLayer
+                    topRight={<PinIndicator />}
+                    topLeft={
+                      hasInitials && showDisabledIcon(pinnedTile.audio) ? (
+                        <DisabledMicIcon isLoading={isLoading(pinnedTile.audio)} />
+                      ) : null
+                    }
+                    bottomLeft={<NameTag name={pinnedTile.displayName} />}
+                  />
+                  <PinTileButton pinned={true} onClick={() => unpin(pinnedTile.mediaPlayerId)} />
```

Comment:
What do you think about renaming it to "PinTileLayer" or "HoverLayer"?

Reply:
`PinTileLayer` is fine. I'll rename it